lib/apex_charts/renderer.rb

```diff
@@ -37,14 +37,30 @@ def defer(js)
         <<~DEFERRED
           (function() {
             var createChart = function() {
+              document.removeEventListener("turbo:load", createChart, true);
+              window.removeEventListener("load", createChart, true);
+
               #{indent(js)}
+              var destroyChart = function() {
+                document.removeEventListener("turbo:before-cache", destroyChart, true);
+                if (#{variable}) {
+                  #{variable}.destroy();
+                  #{variable} = null;
+                }
+              };
+
+              document.addEventListener("turbo:before-cache", destroyChart, true);
```

Comment:
Instead of `before-cache`, shouldn't we handle `before-render`?
This is how Chartkick.js handles it:
https://github.com/ankane/chartkick.js/blob/47edcf182037e943ecb8b09c068fc733d5d68fec/src/index.js#L19-L30

Please handle `turbolinks:` events too.